### PR TITLE
[FEAT] argilla server: add basic endpoints telemetry support

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR adds a middleware component to track the API endpoint's usage.

**Type of change**

- New feature (non-breaking change which adds functionality)
- Refactor (change restructuring the codebase without changing functionality)
- Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla-server/tests/unit/test_endpoints_telemetry.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        settings.enable_telemetry = True
+
+        client = TestClient(create_server_app())
+
+        response = client.post("/api/v1/datasets")
+        assert response.status_code == 401
+
+        test_telemetry.track_endpoint.assert_called_once_with("/api/v1/datasets", ANY, ANY)
+
+    def test_not_track_endpoint_call_with_telemetry_disabled(self, test_telemetry: TelemetryClient):
```
Comment (frascuchon) [inline argilla-server/tests/unit/test_endpoints_telemetry.py]:
```suggestion
    def test_not_track_endpoint_call_with_disabled_telemetry(self, test_telemetry: TelemetryClient):
```

File: argilla-server/src/argilla_server/_app.py [hunk 2]
```diff
@@ -121,6 +124,32 @@ def configure_api_router(app: FastAPI):
     app.mount("/api/v1", api_v1)
 
 
+def configure_telemetry(app: FastAPI):
+    """
+    Configures telemetry middleware for the app if telemetry is enabled
+
+    """
```
Comment (jfcalvo) [inline argilla-server/src/argilla_server/_app.py]:
```suggestion
    """
```

File: argilla-server/src/argilla_server/_app.py [hunk 3]
```diff
@@ -121,6 +124,32 @@ def configure_api_router(app: FastAPI):
...
+    if not settings.enable_telemetry:
+        return
```
Comment (jfcalvo) [inline argilla-server/src/argilla_server/_app.py:132]: What do you think about moving this to the `configure_telemetry(app)` call so we avoid checking it on every request?

Something like:

```python
if settings.enable_telemetry:
    configure_telemetry(app)
```

File: argilla-server/src/argilla_server/_app.py [hunk 4]
```diff
@@ -121,6 +124,32 @@ def configure_api_router(app: FastAPI):
...
+
+    @app.middleware("http")
+    async def track_api_endpoint(request: Request, call_next):
+        response = await call_next(request)
+        try:
+            endpoint_path = resolve_endpoint_path_for_request(request)
```
Comment (jfcalvo) [inline argilla-server/src/argilla_server/_app.py]: Why not moving this logic to be inside the telemetry client? With that you can pass only `request` and `response` like:

```python
await get_telemetry_client().track_endpoint(request, response)
```

File: argilla-server/src/argilla_server/_app.py [hunk 5]
```diff
@@ -121,6 +124,32 @@ def configure_api_router(app: FastAPI):
...
+def configure_telemetry(app: FastAPI):
+    """
+    Configures telemetry middleware for the app if telemetry is enabled
+
+    """
+    if not settings.enable_telemetry:
+        return
+
+    @app.middleware("http")
+    async def track_api_endpoint(request: Request, call_next):
```
Comment (jfcalvo) [inline argilla-server/src/argilla_server/_app.py]: I believe a better naming could be something like:

```suggestion
    async def track_api_request(request: Request, call_next):
```

File: argilla-server/src/argilla_server/_app.py [hunk 6]
```diff
@@ -121,6 +124,32 @@ def configure_api_router(app: FastAPI):
...
+        response = await call_next(request)
+        try:
+            endpoint_path = resolve_endpoint_path_for_request(request)
+
+            if endpoint_path:
+                await get_telemetry_client().track_endpoint(
```
Comment (jfcalvo) [inline argilla-server/src/argilla_server/_app.py]:
```suggestion
                await get_telemetry_client().track_api_request(
```

**Review by jfcalvo — CHANGES_REQUESTED** [review]

File: argilla-server/src/argilla_server/_app.py (hunk 3 above)
Comment (frascuchon) [reply argilla-server/src/argilla_server/_app.py:132]: The `configure_telemetry` function will be called once when FastAPI is created, not for every request. If telemetry is disabled, the telemetry middleware won't be loaded.

File: argilla-server/src/argilla_server/_app.py (hunk 4 above)
Comment (frascuchon) [reply argilla-server/src/argilla_server/_app.py]: Yes, I thought about this when developing it. It makes sense and the middleware keeps simpler. I will push the changes

File: argilla-server/tests/unit/api/handlers/v1/test_datasets.py [hunk 7]
```diff
@@ -209,6 +209,7 @@ async def test_list_current_user_datasets_by_workspace_id(
         response_body = response.json()
         assert [dataset["name"] for dataset in response_body["items"]] == ["dataset-a"]
 
+    @pytest.mark.skip
```
Comment (jfcalvo) [inline argilla-server/tests/unit/api/handlers/v1/test_datasets.py:212]: Why we need to skip these tests?

**Review by jfcalvo — APPROVED** [review]

File: argilla-server/tests/unit/api/handlers/v1/test_datasets.py (hunk 7 above)
Comment (frascuchon) [reply argilla-server/tests/unit/api/handlers/v1/test_datasets.py:212]: I needed to build the docker image. Those tests are failing in the base branch. I review all of them in a separate PR.

**Pull request merged**